File: day1-foundations/1-simple-chat.js
import { getModel } from "../common/model.js";
import { HumanMessage, SystemMessage } from "@langchain/core/messages";

const run = async () => {
    await askQuestion("What is LangGraph?");
    await askQuestion("What is the latest version of LangGraph?");
    await askQuestion("What is LangGraph inspired by?");
};

async function askQuestion(question) {
    const model = getModel();
    let messages = [
        new SystemMessage("You are a helpful assistant."),
        new HumanMessage(question),
    ];

    console.log('\n\n----------------\n' +question);
    let response = await model.invoke(messages);

    console.log("\nResponse:");
    console.log(response.content);
}

run().catch(console.error);

File: day1-foundations/1-simple-chat.py
import sys
import os

# Add the parent directory to sys.path to allow imports from common
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from common.model import get_model
from langchain_core.messages import HumanMessage, SystemMessage
import asyncio

async def run():
    await ask_question("What is LangGraph?")
    await ask_question("What is the latest version of LangGraph?")
    await ask_question("What is LangGraph inspired by?")

async def ask_question(question):
    model = get_model()
    messages = [
        SystemMessage(content="You are a helpful assistant."),
        HumanMessage(content=question),
    ]

    print('\n\n----------------\n' + question)
    response = await model.ainvoke(messages)

    print("\nResponse:")
    print(response.content)

if __name__ == "__main__":
    asyncio.run(run())
